lib/alephant/harness/service/s3.rb

```diff
@@ -1,60 +1,50 @@
-require 'aws-sdk'
+require "aws-sdk-s3"
 
 module Alephant
   module Harness
     module Service
       module S3
-
-        def self.client
-          @@client ||= ::AWS::S3.new
-        end
-
         def self.create(id)
-          client.buckets.create id
+          client.create_bucket(bucket: id)
         end
 
         def self.delete(id)
-          client.buckets[id].tap do |bucket|
-            bucket.objects.each do |object|
-              object.delete
-            end
-            bucket.delete
-          end
+          s3 = Aws::S3::Resource.new(client: client)
+          s3.bucket(id).delete
         end
 
         def self.add_object(id, object_id, data)
-          client.buckets[id]
-                .objects[object_id]
-                .write(data)
+          client.put_object(
+            body: data,
+            bucket: id,
+            key: object_id
+          )
         end
 
         def self.get_object(id, object_id)
-          client.buckets[id]
-                .objects[object_id]
-        end
-
-        def self.delete_object(id, object_id)
-          get_object(id, object_id).delete
+          client.get_object(
+            bucket: id,
+            key: object_id
+          )
         end
 
         def self.bucket_exists?(bucket_id)
           begin
-            exists = client.buckets[bucket_id].exists?
-          rescue ::AWS::S3::Errors::NoSuchKey => e
-            exists = false
-          end
-
-          exists.tap do |e|
-            yield if e && block_given?
+            client.head_bucket(
+              bucket: bucket_id
+            )
+            yield if block_given?
+            true
+          rescue => e
+            false
           end
         end
 
-        def self.exists?(id, object_id)
-          if get_object(id, object_id)
-            yield
-          end
-        end
+        private
 
+        def self.client
+          @client ||= ::Aws::S3::Client.new(AWS.s3_config)
```

Comment:
this used to be a cvar `@@client`, and in the other services it still is. Was this change intentional?